sess.go

```diff
@@ -20,29 +17,23 @@ var allSessions sync.Map
 // see: https://www.postgresql.org/docs/9.2/static/protocol.html
 // for postgres protocol and startup handshake process
 type session struct {
-	Server      *server
-	Conn        net.Conn
-	Args        map[string]interface{}
-	Secret      int32 // used for cancelling requests
-	Ctx         context.Context
-	CancelFunc  context.CancelFunc
-	initialized bool
-}
-
-// implements Queryer
-func (s *session) Query(ctx context.Context, n nodes.Node) (driver.Rows, error) {
-	return s.Server.Query(ctx, n)
-}
-
-// implements Execer
-func (s *session) Exec(ctx context.Context, n nodes.Node) (driver.Result, error) {
-	return s.Server.Exec(ctx, n)
+	Server        *server
+	Conn          io.ReadWriteCloser
+	Args          map[string]interface{}
+	Secret        int32 // used for cancelling requests
+	Ctx           context.Context
+	CancelFunc    context.CancelFunc
+	initialized   bool
+	queryer       Queryer
+	authenticator authenticator
 }
 
 // Handle a connection session
 func (s *session) Serve() error {
-	// read the initial connection startup message
-	msg, err := s.Read()
+
```

Comment:
not a big deal, but we are not leaving empty line at the begging of functions (applies to all funcs with empty line in this pr 😄)